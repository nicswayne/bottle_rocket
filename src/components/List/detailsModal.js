import React from 'react';
import MapComponent from './map';

const detailsModal = ({ google }) => (
    <div>
        <MapComponent
            google={google}
            zoom={14}
            containerStyle={{
                height: '180px',
                width: '100%',
                position: 'relative',
            }}
            location={{
                lat: 32.950787,
                lng: -96.821118,
            }}
            name="test location"
        />
        <div className="details-header">
            <div className="name text">
                <strong>NAME</strong>
                <div className="category">CATEGORY</div>
            </div>
        </div>
        <div className="details">
            <div className="name pad-16">Address</div>
            <div className="name pad-26">phone</div>
            <div className="name pad-26">twitter</div>
        </div>
    </div>
);

export default detailsModal;
