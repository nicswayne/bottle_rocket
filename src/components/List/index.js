import React from 'react';

const List = () => <div>List page</div>;

export default List;
