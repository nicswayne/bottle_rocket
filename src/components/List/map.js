import React from 'react';
import { Map, Marker } from 'google-maps-react';

const MapComponent = ({ google, location, containerStyle, zoom, name }) => (
    <div className="loc-map">
        <div className="map-content">
            <Map
                google={google}
                zoom={zoom}
                clickableIcons={false}
                initialCenter={location}
                containerStyle={containerStyle}
            >
                <Marker title={name} name={name} position={location} />
            </Map>
        </div>
    </div>
);

export default MapComponent;
