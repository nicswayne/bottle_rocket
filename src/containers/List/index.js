import { connect } from 'react-redux';

import { getList } from '../../actions/restaurants';

import List from '../../components/List';

const mapDispatchToProps = dispatch => ({
    getList: () => dispatch(getList()),
});

export default connect(null, mapDispatchToProps)(List);
