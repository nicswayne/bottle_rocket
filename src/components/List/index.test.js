import React from 'react';
import { shallow, render } from 'enzyme';

import List from '.';

it('List component should render without crashing.', () => {
    shallow(<List getList={jest.fn()} />);
});

it('List component should match snapshot.', () => {
    const wrapper = render(<List getList={jest.fn()} />);

    expect(wrapper).toMatchSnapshot();
});
