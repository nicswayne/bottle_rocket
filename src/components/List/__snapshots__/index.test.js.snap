// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component should match snapshot. 1`] = `
<div>
  <div>
    <div
      class="header"
    >
      <div
        class="text"
      >
        <strong>
          Lunch Tyme
        </strong>
        <img
          alt=""
          class="map"
          src="icon_map@2x.png"
        >
          
        </img>
      </div>
    </div>
    <div
      class="list-body"
    >
      
    </div>
    <div
      class="footer"
    >
      <a
        class="button"
        href="/"
      >
        <img
          alt=""
          src="tab_lunch@2x.png"
        >
          
        </img>
        <div
          class="text"
        >
          lunch
        </div>
      </a>
      <a
        class="button"
        href="/web"
      >
        <img
          alt=""
          src="tab_internets@2x.png"
        >
          
        </img>
        <div
          class="text"
        >
          internets
        </div>
      </a>
    </div>
  </div>
  <div>
    
  </div>
</div>
`;
